src/store/user/profile.ts

```diff
@@ -0,0 +1,20 @@
+import {createSlice, PayloadAction} from '@reduxjs/toolkit';
+import {Dict} from 'types/store';
+import {USER_PROFILE} from 'constants/redux';
+
+const profile = createSlice({
+  initialState: {} as Dict<any>,
+  name: USER_PROFILE,
+  reducers: {
+    setProfile: (state, {payload}: PayloadAction<any>) => {
+      const {name, account, githubUsername} = payload;
+      state.name = name;
+      state.account = account;
+      state.githubUsername = githubUsername;
```

Comment:
So with these slices, its not required to create a copy of state and we can directly mutate it ?. Is there internally something handling this ?

Reply:
yeah, immer does it.